src/main/java/gov/cms/ab2d/eventlogger/api/EventsListener.java

```diff
@@ -1,31 +1,49 @@
 package gov.cms.ab2d.eventlogger.api;
 
+import gov.cms.ab2d.eventclient.messages.AlertSQSMessage;
 import gov.cms.ab2d.eventclient.messages.GeneralSQSMessage;
+import gov.cms.ab2d.eventclient.messages.KinesisSQSMessage;
+import gov.cms.ab2d.eventclient.messages.LogAndTraceSQSMessage;
 import gov.cms.ab2d.eventclient.messages.SQSMessages;
+import gov.cms.ab2d.eventclient.messages.SlackSQSMessage;
+import gov.cms.ab2d.eventclient.messages.TraceAndAlertSQSMessage;
+import gov.cms.ab2d.eventclient.messages.TraceSQSMessage;
 import gov.cms.ab2d.eventlogger.LogManager;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.cloud.aws.messaging.listener.SqsMessageDeletionPolicy;
 import org.springframework.cloud.aws.messaging.listener.annotation.SqsListener;
+import org.springframework.context.annotation.DependsOn;
 import org.springframework.stereotype.Service;
 
 @Slf4j
 @Service
+@DependsOn({"SQSConfig"})
 public class EventsListener {
-    private LogManager logManager;
+    private final LogManager logManager;
 
     public EventsListener(LogManager logManager) {
         this.logManager = logManager;
     }
 
-    @SqsListener(value = "ab2d-events", deletionPolicy = SqsMessageDeletionPolicy.ON_SUCCESS)
+    @SqsListener(value = "${sqs.queue-name}", deletionPolicy = SqsMessageDeletionPolicy.ON_SUCCESS)
     public void processEvents(SQSMessages sqsMessage) {
-        log.info("EventsListener: Processing events from SQS: " + sqsMessage);
+        log.info("EventsListener: Processing events from SQS: " + sqsMessage.getClass().getSimpleName());
         switch (sqsMessage.getClass().getSimpleName()) {
-            case "GeneralSQSMessage":
-                logManager.log(((GeneralSQSMessage) sqsMessage).getLoggableEvent());
-                break;
-            default:
-                log.info("Can't Identify Message");
+            case "GeneralSQSMessage" ->
```

Comment:
There should be a better way to do this that doesn't rely on strings. For example, switch the actual class, create a superclass with either getMessage or getLogabble event and getEnvironments. Maybe also an enum.

Reply:
The tricky part is that the message type determines what to do. So even if we change this, we will still have to do these checks. At least the string comes from the getClass method. Also, switches don't support class check.